torchtext/data/dataset.py

```diff
@@ -197,6 +197,19 @@ def download(cls, root, check=None):
                             shutil.copyfileobj(gz, uncompressed)
 
         return os.path.join(path, cls.dirname)
+ 
+
+    def filter_examples(self, field_names):
+        """Remove unknown words from dataset examples with respect to given field.
+
+        Arguments:
+            field_names (list(str)): Within example only the parts with field names in
+                field_names will have their unknown words deleted.
+        """
+        for i, example in enumerate(self.examples):
+            example_part = [word for word in getattr(example, field_name) if word in self.fields[field_name].vocab.stoi]
```

Comment:
`stoi` is a list, so it would be good to cast it to a set (this can be done outside of the list comprehension otherwise this line is going to explode

Reply:
Done